src/modules/queue/QueueComponent.jsx

```diff
@@ -21,79 +18,61 @@ class QueueContainer extends React.Component {
 	};
 
 	render() {
-		var queue = {};
-		if (this.state.queueOpen) {
-			queue.open = ' open';
-		} else {
-			queue.open = '';
+		if (!this.state.queueOpen) {
+			return (
+				<Positioning>
+					<QueueButton onClick={this.toggleQueue}>
+						<img src={queue_img} />
+					</QueueButton>
+				</Positioning>
+			);
 		}
 		return (
-			<QueueHolder className={queue.open}>
-				<div className="queue-list">
+			<Positioning>
+				<QueueList>
 					<QueueItem>
 						<div className="item-image" />
 						<div className="item-info">
 							<h4>Song Title</h4>
 						</div>
 					</QueueItem>
-				</div>
-				<QueueButton onClick={this.toggleQueue}>
-					<img src={queue_img} />
-					<div className="queue-x" />
-				</QueueButton>
-			</QueueHolder>
+					<QueueButton onClick={this.toggleQueue} className="open">
+						<img src={x_img} />
+					</QueueButton>
+				</QueueList>
+			</Positioning>
 		);
 	}
 }
 
-const QueueHolder = styled.div`
-	display: inline-block;
+const Positioning = styled.div`
 	position: absolute;
-	right: 42px;
+	bottom: 0;
+	right: 2.625rem;
 	z-index: 10;
-
-	&.open {
-		.queue-list {
-			display: inline-block;
-		}
-		img {
-			opacity: 0;
-		}
-		.queue-x {
-			opacity: 1;
-		}
-	}
-
-	.queue-list {
-		background: rgba(255, 255, 255, 0.9);
-		bottom: -24px;
-		display: none;
-		height: 600px;
-		position: absolute;
-		right: -24px;
-		min-width: 320px;
-		width: 25%;
-		z-index: 1;
-	}
 `;
 
+const QueueList = styled.div`
+	background: rgba(255, 255, 255, 0.9);
+	min-height: 37.5rem;
```

Comment:
Favor `min-height` over `height`. If we add more and more items, `min-height` will allow the container to grow when the children exceed the height, `height` will just hide all the overflow.

Also note that I'm using rem units instead of px. Don't really worry about that, it's just habit for me.